Geometry/TrackerNumberingBuilder/src/GeometricDet.cc

```diff
@@ -73,7 +73,7 @@ GeometricDet::GeometricDet(DDFilteredView* fv, GeometricEnumType type)
       _phi(_trans.Phi()),
       _rho(_trans.Rho()),
       _rot(fv->rotation()),
-      _shape(fv->shape()),
+      _shape(cms::DDSolidShape(static_cast<int>(fv->shape()))),
```

Comment:
see my previous comment - using an `int` would remove the need for casts. It would correspond to an index of a `cms::DDSolidShape`.

Reply:
In theory yes, but then we'd need to have a direct correspondence between the `DDSolidShape` and `cms::DDSolidShape` indices, right? (note: the same is true for the cast here, so in any case this bit of code is inconsistent - probably the safest is to always use the map between the two)

Reply:
yes, the map between the two already exists. Here is a simple test:
```c++
Running .
Old id: 0 == new id: 0
Old id: 1 == new id: 1
Old id: 2 == new id: 2
Old id: 3 == new id: 3
Old id: 4 == new id: 4
Old id: 5 == new id: 5
Old id: 7 == new id: 5
Old id: 6 == new id: 6
Old id: 8 == new id: 6
Old id: 9 == new id: 7
Old id: 10 == new id: 8
Old id: 11 == new id: 9
Old id: 12 == new id: 10
Old id: 13 == new id: 11
Old id: 14 == new id: 12
Old id: 15 == new id: 13
Old id: 16 == new id: 14
Old id: 17 == new id: 15
Old id: 18 == new id: 16
Old id: 19 == new id: 17

OK (1)

```

Reply:
Ok - so what I'll do is put the enum back to the way it was, and change the cast here to instead use the mapping explicitly

Reply:
@adewit Yes, that is good plan.